pkg/process/maps.go

```diff
@@ -318,65 +315,81 @@ func (m *Mapping) findProgramHeader(ef *elf.File, addr uint64) (*elf.ProgHeader,
 	return elfreader.HeaderForFileOffset(headers, addr-uint64(m.StartAddr)+uint64(m.Offset))
 }
 
-func (m *Mapping) computeBase(ef *elf.File, addr uint64) (uint64, error) {
+func (m *Mapping) extractExecutableInfo(ef *elf.File, addr uint64) (*profilestorepb.ExecutableInfo, error) {
 	if m == nil {
-		return 0, nil
+		return nil, nil
 	}
 	if addr < uint64(m.StartAddr) || addr >= uint64(m.EndAddr) {
-		return 0, fmt.Errorf("specified address %x is outside the mapping range [%x, %x]", addr, m.StartAddr, m.EndAddr)
+		return nil, fmt.Errorf("specified address %x is outside the mapping range [%x, %x]", addr, m.StartAddr, m.EndAddr)
 	}
 
 	loadSegment, err := m.findProgramHeader(ef, addr)
 	if err != nil {
-		return 0, fmt.Errorf("failed to find program header for mapping %#v: %w", m, err)
+		return nil, fmt.Errorf("failed to find program header for mapping %#v: %w", m, err)
 	}
 
-	base, err := elfreader.Base(
```

Comment:
Since we move this method to upstream Parca, shall we clean it up? Is it used anywhere else?

Reply:
no you're right!